### PR TITLE
Offer inbox RPC

### PR DESCRIPTION
**Pull request by rorp**

### CODE REVIEW

File: docs/applications/server.md [hunk 1]
```diff
@@ -300,6 +300,13 @@ the `-p 9999:9999` port mapping on the docker container to adjust for this.
  - `getdlcs` - Returns all dlcs in the wallet
  - `getdlc` `dlcId` - Gets a specific dlc in the wallet
     - `dlcId` - Internal id of the DLC
+ - `registerincomingoffer` `offerTLV` `peer` `message` - Puts an incoming offer into the inbox
```
Comment (user411) [inline docs/applications/server.md]: Might want a standardized name on these like 'offer-add', 'offer-send', 'offer-remove', 'offers-list'.

File: core/src/main/scala/org/bitcoins/core/api/dlc/wallet/db/IncomingDLCOfferDb.scala [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+import java.time.Instant
+
+case class IncomingDLCOfferDb(
+    hash: Sha256Digest,
+    receivedAt: Instant,
+    peer: Option[String],
+    message: Option[String],
+    offerTLV: DLCOfferTLV) {
+  require(peer.forall(_.length <= 1024),
```
Comment (user411) [inline core/src/main/scala/org/bitcoins/core/api/dlc/wallet/db/IncomingDLCOfferDb.scala:14]: Might want to flip the max length on these. More likely to want a long message than a 1024 character tor address.

File: app/server/src/main/scala/org/bitcoins/server/ServerJsonModels.scala [hunk 3]
```diff
@@ -1358,6 +1359,49 @@ object CreateContractInfo extends ServerJsonModels {
...
+      case offerTlvJs :: peerJs :: messageJs :: Nil =>
+        Try {
+          val tlv = DLCOfferTLV.fromHex(offerTlvJs.str)
+          val peer = nullToOpt(peerJs).map(_.str)
+          val message = nullToOpt(messageJs).map(_.str)
+          RegisterIncomingOffer(tlv, peer,  message)
+        }
+      case other =>
+        val exn = new IllegalArgumentException(
+          s"Bad number or arguments to registerincomingoffer, got=${other.length} expected=1")
```
Comment (user411) [inline app/server/src/main/scala/org/bitcoins/server/ServerJsonModels.scala]: expected=3

**Comment by user411** [comment]
So is the `register` endpoint essentially `save`? There is no `send` endpoint right now? So there is save/remove/list IncomingOffer endpoints?

**Comment by Christewart** [comment]
Looks like latest commit breaks compile

**Review by Christewart — APPROVED** [review]
LGTM, would like an approval from @user411 before merging

**Review by user411 — APPROVED** [review]

**Pull request merged**